clause/label/label.go

```diff
@@ -0,0 +1,42 @@
+package label
```

Comment:
This is the one clause type I've added, the most obvious bit of searching on a filename. Right now it's still sort of a stub -- our current code splits up the string, adds wildcards, and joins with OR, where this just passes the string through exactly. Ultimately, there should also be an extra argument for whether or not to search exactly, and perhaps others to control the behavior. For now, this is mostly just an example of how the pluggable clause system works.

Reply:
Each clause type will end up getting it's own unit test, right?

Reply:
That's the plan. I haven't written any yet because I haven't sat down and decided what makes sense to be within the scope to test there, though I suppose at least simple "input provides expected output" sorts of things might be all.

Reply:
Like I mentioned below, arg parsing would be something to test. 
That seems highly dependent on the clause.

Reply:
Hm, yeah, that's a good idea. I think once I'm adding more clauses I might start adding utility functions for that which make sense to be tested in isolation (I'm liking how `mapstructure` works, I'll probably use that for most/all of them), which might also do various sorts of validation etc. if I need to add any of that.